### PR TITLE
Use isVisible flag for modal open close

### PR DESCRIPTION
**Pull request by ddc22**
Related to #

## Proposed Changes

* Uses the Dialog's isVisible variable instead of unmounting the entire component

## Testing Instructions

* Go through `/start/onboarding-pm`
* Select a free domain
* Select the free plan
* Make sure the modal appears.
* Try the close button and make sure the modal disappears.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/plans-features-main/components/plan-upsell-modal/index.tsx [hunk 1]
```diff
@@ -108,7 +108,7 @@ export default function ModalContainer( props: ModalContainerProps ) {
 	return (
 		<Dialog
 			isBackdropVisible={ true }
-			isVisible={ true }
+			isVisible={ isModalOpen && !! modalType }
```
Comment (chriskmnds) [inline client/my-sites/plans-features-main/components/plan-upsell-modal/index.tsx:106]: I'm guessing we'd need to remove the lines above, right?

```
	/** Explore using the isVisible flag on Dialog instead. */
	if ( ! isModalOpen || ! modalType ) {
		return;
	}
```

**Review by chriskmnds — COMMENTED** [review]
Thanks for following up. I left the comment in the code.

The other thing to make sure of (I think) is that `Global` is unmounted from Dialog when closing (I think you've confirmed this already right?). Otherwise, it sounds like we'd have those styles linger onto any other modals on the page.

File: client/my-sites/plans-features-main/components/plan-upsell-modal/index.tsx (hunk 1 above)
Comment (ddc22) [reply client/my-sites/plans-features-main/components/plan-upsell-modal/index.tsx:106]: Ah yes seems I forgot to push a local commit :D

**Comment by ddc22** [comment]
> The other thing to make sure of (I think) is that Global is unmounted from Dialog when closing

Interested point 😮 !!!! I investigated this and it does. And TIL `ReactDOM.unmountComponentAtNode(this.node)`

https://github.com/reactjs/react-modal/blob/master/src/components/Modal.js#L193

This seems to be triggering the [unmount](https://github.com/emotion-js/emotion/blob/f3b268f7c52103979402da919c9c0dd3f9e0e189/packages/react/src/global.js#L113) in emotion `Global`

**Review by chriskmnds — APPROVED** [review]
works as outlined :shipit:

**Pull request merged**